Codeinwp/otter-blocks
# Add live search

Closes #1135, https://github.com/Codeinwp/otter-internals/issues/57.


### Summary

Adds a block variant to the existing `core/search` block named `Live Search`.

This will add two additional attributes: `isLive` and `postTypes` that can be set from the inspector under the `Live Search` panel.

If a search block has live search enabled, in the frontend when searching it will look like this

<img width="650" alt="Screenshot 2022-11-30 at 12 25 45" src="https://user-images.githubusercontent.com/39873395/204772001-1ca7cc99-1bf4-43f2-ba79-d102a2891bdb.png">


When clicking the `Search` button, it will behave like the ordinary search block.

### Test instructions

- Add the Search block or the Live Search variant. Both of them should have an additional panel in the inspector.
- The Live Search one should have the `Enable Live Search` toggle on by default.
- Add different post types in the Token Field and check if the results are filtered accordingly in the frontend.
- The results container should close when clicking outside the search block or when the input is emptied
- Using the arrow keys up & down, you can navigate through the results and when pressing Enter, the selected result should open.



---- 

### Checklist before the final review
